File: Leetcode/array/TwoSum/src/cpp/TwoSum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int> hash;
        vector<int> ret;
        for(int i=0;i<nums.size();i++){
            int numberToFind=target-nums[i];
            if(hash.find(numberToFind)!=hash.end()){
                ret.push_back(hash[numberToFind]);
                ret.push_back(i);
                return ret;
            }
            hash[nums[i]]=i;
        }
        return ret;
    }
};

File: Leetcode/array/TwoSum/src/python/TwoSum.py
class Solution(object):
    def twoSum(self,nums,target):
        if len(nums)<=1:
            return False
        buff_dict={}
        for i in xrange(len(nums)):
            if nums[i] in buff_dict:
                return [buff_dict[nums[i]],i]
            else:
                buff_dict[target-nums[i]]=i

if __name__ =="__main__":
    print Solution().twoSum((3,5,9,13,15),20)